src/bluesearch/database/download.py

```diff
@@ -213,6 +215,67 @@ def get_s3_urls(
     return url_dict
 
 
+def get_gcs_urls(
+    bucket: Bucket,
+    start_date: datetime,
+    end_date: datetime | None = None,
+) -> dict[str, list[Blob]]:
+    """Get Google Cloud Storage urls.
+
+    Parameters
+    ----------
+    bucket
+        GCS bucket.
+    start_date
+        Starting date to download the incremental files (inclusive).
+    end_date
+        Ending date. If None, today is considered as the ending date (inclusive).
+
+    Returns
+    -------
+    url_dict
+        Keys represent different months. Values are list of blobs
+        corresponding to actual PDF files.
+    """
+    date_list = get_daterange_list(
+        start_date=start_date,
+        end_date=end_date,
+        delta="month",
+    )
+    yearmonth_list = [date.strftime("%y%m") for date in date_list]
+
+    client = bucket.client
+
+    url_dict = {}
+    for yearmonth in yearmonth_list:
+        iterator = client.list_blobs(bucket, prefix=f"arxiv/arxiv/pdf/{yearmonth}")
+
+        # If more than one version is found, we only keep the last one
+        df = pd.DataFrame(
+            (
+                (
+                    el,
+                    el.name,
+                    el.name.rsplit("v", 1)[0],
+                    int(el.name.rsplit("v", 1)[1].split(".")[0]),
+                )
+                for el in iterator
+            ),
+            columns=["blob", "fullname", "article", "version"],
+        )
+
+        df_latest = df[["article", "version"]].groupby("article", as_index=False).max()
+        fullname_latest = df_latest["name"] = (
+            df_latest["article"] + "v" + df_latest["version"].astype(str) + ".pdf"
+        )
+
+        url_dict[yearmonth] = df.loc[
+            df["fullname"].isin(fullname_latest), "blob"
+        ].to_list()
```

Comment:
Looks fine to me. Just to note that the original `HTTPIterator` is lazy (I think), so evaluating it might lead to a slowdown. But maybe there's no other way.

Not a must, but here's an alternative implementation without pandas:
```python
    import re

    pattern = re.compile(r"(.*)v(\d+).pdf")
    for yearmonth in yearmonth_list:
        blobs = {}
        versions = {}
        for blob in client.list_blobs(bucket, prefix=f"arxiv/arxiv/pdf/{yearmonth}"):
            # Assuming pattern will always match
            name, version = pattern.fullmatch(blob.name).groups()
            if name not in versions or versions[name] < int(version):
                blobs[name] = blob
                versions[name] = int(version)
        url_dict[yearmonth] = list(blobs.values())
```

Reply:
Unrolling the `HTTPIterator` certainly isn't ideal, but I also cannot see another way to do that... anyway it runs in just a few seconds, so hopefully it's not too much of a big deal.

I went with the `pandas` solution as I hoped in some speedup, but if you prefer I can switch to the `dict` solution you propose!